src/rust/scheduler/scheduler.rs

```diff
@@ -339,7 +339,7 @@ mod tests {
             None => anyhow::bail!("insert() failed"),
         };
         let task_id2: u64 = handle2.get_task_id();
-        crate::ensure_neq!(task_id2, task_id + 1);
+        crate::ensure_neq!(task_id2, task_id);
```

Comment:
Thanks for fixing that. Important question is: why the test was passing previously?

Reply:
I believe it was just a matter of time that it would fail... :)

Reply:
I also wanted to check that the scheduler is not allocating task IDs sequentially. I suppose even with a random allocation, this might fail occasionally. Any better ideas for how to test this? Anand was the one that suggested that I write a test case for this.  :)

Reply:
Oh, I was just asking for a test covering this code :).

If we stick to a good/decent random number generator (RNG) then we don't need to be worried about this at all because that's a test for the RNG, and not for this code.

Reply:
> I believe it was just a matter of time that it would fail... :)

It will not fail if and only if the number generator is somehow is generating numbers in sequence. We can check that by changing the seed number.  Could you verify? I would say that there is something else going on here.

Reply:
I didn't understand your point. This test will only fail (in the current state without my change) if the 2 task ids are in sequence. It has nothing to do with the seed. Which scenario are you thinking about?

Reply:
I think Pedro is wondering if the random number generator ever generates two numbers in sequence for a given seed and if it does, we should just use a different seed. I think it's pretty unlikely but not impossible that the RNG produces to numbers in sequence, so we should probably not test it. I agree with this change.